modcc/lexer.cpp

```diff
@@ -195,6 +195,20 @@ Token Lexer::parse() {
                 }
                 return t;
             }
+            case '&':
+                t.spelling += character();
+                if(*current_=='&') {
+                    t.spelling += character();
+                    t.type = tok::and_t;
+                    return t;
+                }
```

Comment:
What happens if there is a single `&` or `|`, e.g. `if (foo & bar)` ?

In that case it should be an error, i.e. every `&` should be followed by another `&`, right?

```c++
t.spelling += character();
if(*current_=='&') {
    t.spelling += character();
    t.type = tok::and_t;
    return t;
}
else {
    error_string_ = pprintf("Unknown symbol '&': did you mean '&&'?");
    status_ = lexerStatus::error;
    t.type = tok::reserved;
    return t;
}
```

Reply:
True, I'll add tests and make sure errors are thrown when needed.